src/li_chao.rs

```diff
@@ -0,0 +1,110 @@
+/// A structure for answering maximum queries on a set of linear functions. Supports two
+/// operations: inserting a linear function and querying for maximum at a given point. Unlike the
+/// simplest convex hull trick implementation, the queries can be done in any order, and we can do
```

Comment:
I'm curious, what's the simple implementation? I don't think the one in `order.rs` has this limitation; it might be more useful to explicitly compare against that version.

Reply:
By the way, it's not necessary for the nodes to be consecutive integers is it? As long as we know all the query locations (`x`s) in advance...

Reply:
The simplest version doesn't do any sqrt decomposition or anything like that, it just adds lines in increasing-slope order. I'm not as familiar with your algorithm, but I can change the wording here.